lib/Compiler.php

```diff
@@ -0,0 +1,195 @@
+        $finder->files()
+            ->ignoreVCS(true)
+            ->name('*.php')
+            ->name('LICENSE')
+            ->exclude('Tests')
+            ->exclude('tests')
+            ->in(__DIR__.'/../vendor/composer/')
+            ->in(__DIR__.'/../vendor/doctrine/annotations')
+            ->in(__DIR__.'/../vendor/doctrine/inflector')
+            ->in(__DIR__.'/../vendor/doctrine/lexer')
+            ->in(__DIR__.'/../vendor/goetas/')
+            ->in(__DIR__.'/../vendor/jms/')
+            ->in(__DIR__.'/../vendor/phpcollection/')
+            ->in(__DIR__.'/../vendor/phpoption/')
+            ->in(__DIR__.'/../vendor/symfony/')
+            ->in(__DIR__.'/../vendor/zendframework/')
+        ;
+
+        foreach ($finder as $file) {
+            $this->addFile($phar, $file);
+        }
+
+        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/autoload.php'));
+        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/composer/autoload_namespaces.php'));
+        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/composer/autoload_psr4.php'));
+        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/composer/autoload_classmap.php'));
+        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/composer/autoload_real.php'));
+        if (file_exists(__DIR__.'/../vendor/composer/include_paths.php')) {
+            $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/composer/include_paths.php'));
+        }
+        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/composer/ClassLoader.php'));
+        $this->addBin($phar);
+
+        // Stubs
+        $phar->setStub($this->getStub());
+
+        $phar->stopBuffering();
+
+        // disabled for interoperability with systems without gzip ext
+        // $phar->compressFiles(\Phar::GZ);
+
+//        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../../../LICENSE'), false);
+
+        unset($phar);
+    }
+
+    private function addFile(\Phar $phar, \SplFileInfo $file, $strip = true)
+    {
+        $path = strtr(str_replace(dirname(__DIR__).DIRECTORY_SEPARATOR, '', $file->getRealPath()), '\\', '/');
+
+        $content = file_get_contents($file);
+        if ($strip) {
+            $content = $this->stripWhitespace($content);
+        } elseif ('LICENSE' === basename($file)) {
+            $content = "\n".$content."\n";
+        }
+
+        $phar->addFromString($path, $content);
+    }
+
+    private function addBin(\Phar $phar)
+    {
+        $content = file_get_contents(__DIR__.'/../bin/xsd2php');
+        $content = preg_replace('{^#!/usr/bin/env php\s*}', '', $content);
+        $phar->addFromString('bin/xsd2php', $content);
+    }
+
+    /**
+     * Removes whitespace from a PHP source string while preserving line numbers.
+     *
+     * @param  string $source A PHP string
+     * @return string The PHP string with the whitespace removed
+     */
+    private function stripWhitespace($source)
```

Comment:
Just to keep file size small.
See https://github.com/composer/composer/blob/master/src/Composer/Compiler.php#L191 for the origin of the method.
I also use it in my phar compiler classes, it does not break things and even preserves line numbers hence keeping stack traces compatible to a non phar version.